### PR TITLE
Stricter typing and linting

### PR DESCRIPTION
**Pull request by boudra**
- Enable stricter Eslint rules for typing
- Implement all types for passport, calculator and tests

There are no changes to functionality.

depends on https://github.com/gitcoinco/allo-indexer/pull/112

### CODE REVIEW

**Comment by vacekj** [comment]
@boudra what's the status on this? I'm down to tag-team on this to get it merged.

**Comment by boudra** [comment]
> @boudra what's the status on this? I'm down to tag-team on this to get it merged.

need to merge this one first: https://github.com/gitcoinco/allo-indexer/pull/112

then this one is ready to go as well

**Review by vacekj — APPROVED** [review]

File: src/config.ts [hunk 1]
```diff
@@ -20,7 +19,7 @@ const chains: Chain[] = [
   {
     id: 1,
     name: "mainnet",
-    rpc: `https://mainnet.infura.io/v3/${process.env.INFURA_API_KEY}`,
+    rpc: `https://mainnet.infura.io/v3/${process.env.INFURA_API_KEY ?? ""}`,
```
Comment (bard) [inline src/config.ts:22]: Not for this PR but I think that eventually we should have a `getChains` method that returns this object and that throws when the necessary API KEYs aren't provided, so we don't start the service with bad configuration.

**Review by bard — APPROVED** [review]

**Review by vacekj — APPROVED** [review]

File: src/config.ts (hunk 1 above)
Comment (boudra) [reply src/config.ts:22]: @bard that's a great idea, just a note though is that both the Infure and Alchemy API keys are not required, the indexer should still work, might get some more rate limit errors though, but could be worth making them required.

**Pull request merged**